contrib/go-chi/chi/chitrace_test.go

```diff
@@ -0,0 +1,142 @@
+	r.Use(Middleware)
+
+	r.Get("/test/{id}", func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(200)
+		fmt.Fprintln(w, "ᕕ( ᐛ )ᕗ")
+	})
+
+	req := httptest.NewRequest("GET", "/test/24601", nil)
+	res := httptest.NewRecorder()
+	r.ServeHTTP(res, req)
+
+	assert.NotNil(latestSpan)
+	assert.Equal(latestSpan.OperationName, "http.request")
+	assert.Equal(latestSpan.Tags[string(ext.HTTPStatusCode)], "200")
+	assert.Equal(latestSpan.Tags[string(ext.HTTPMethod)], "GET")
+	assert.Equal(latestSpan.Tags[string(ext.HTTPUrl)], "/test/24601")
+	assert.Equal(latestSpan.Tags[dd.ResourceName], "GET /test/{id}")
+	assert.Nil(latestSpan.Tags[string(ext.Error)])
+}
+
+func TestTracingMiddleWareError(t *testing.T) {
+	assert := assert.New(t)
+
+	// Tracing setup
+	tt := newTestTracer()
+	opentracing.SetGlobalTracer(tt)
+
+	// Router setup
+	r := chi.NewRouter()
+	r.Use(Middleware)
+
+	r.Get("/test/{id}", func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(500)
+		fmt.Fprintln(w, "¯\\_(ツ)_/¯")
+	})
+
+	req := httptest.NewRequest("GET", "/test/24601", nil)
+	res := httptest.NewRecorder()
+	r.ServeHTTP(res, req)
+
+	assert.NotNil(latestSpan)
+	assert.Equal(latestSpan.Tags[string(ext.HTTPStatusCode)], "500")
+	assert.Equal(latestSpan.Tags[string(ext.Error)], true)
+}
+
+// Fake Tracer, Span, and Context for test purposes
+// Based on https://github.com/opentracing/opentracing-go/blob/master/testtracer_test.go
+type testSpan struct {
+	spanContext   testSpanContext
+	OperationName string
+	Tags          map[string]interface{}
+}
+
+func (s testSpan) SetTag(key string, value interface{}) opentracing.Span {
+	s.Tags[key] = value
+	return s
+}
+
+func (s testSpan) Equal(os opentracing.Span) bool                         { return true }
+func (s testSpan) Context() opentracing.SpanContext                       { return s.spanContext }
+func (s testSpan) Finish()                                                {}
+func (s testSpan) FinishWithOptions(opts opentracing.FinishOptions)       {}
+func (s testSpan) LogFields(fields ...log.Field)                          {}
+func (s testSpan) LogKV(kvs ...interface{})                               {}
+func (s testSpan) SetOperationName(operationName string) opentracing.Span { return s }
+func (s testSpan) Tracer() opentracing.Tracer                             { return testTracer{} }
+func (s testSpan) SetBaggageItem(key, val string) opentracing.Span        { return s }
+func (s testSpan) BaggageItem(key string) string                          { return "" }
+func (s testSpan) LogEvent(event string)                                  {}
+func (s testSpan) LogEventWithPayload(event string, payload interface{})  {}
+func (s testSpan) Log(data opentracing.LogData)                           {}
+
+type testSpanContext struct {
+	HasParent bool
+	FakeID    int
+}
+
+func (n testSpanContext) ForeachBaggageItem(handler func(k, v string) bool) {}
+
+type testTracer struct{}
+
+func newTestTracer() testTracer {
```

Comment:
This constructor isn't needed. Can you please remove it? You can just the following in your code:
```go
var tt testTracer
```
It's nicer and works the same.